src/components/TextInput/TextInput.tsx

```diff
@@ -25,6 +32,7 @@ export interface TextInputProps extends HTMLAttributes<HTMLInputElement> {
   startAddon?: ReactNode;
   startIcon?: IconNamesType;
   value?: number | string;
+  type?: string;
```

Comment:
recommend enforcing this to be only the types we allow
```suggestion
  type?: "password" | "text" | "number";
```

[See here for the full list of allowed types](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#input_types) if you think that there should be more/fewer.